cmd/juju/controller/destroy.go

```diff
@@ -463,6 +468,8 @@ type destroyCommandBase struct {
 	api       destroyControllerAPI
 	apierr    error
 	clientapi destroyClientAPI
+
+	controllerCredentialAPIF func() (CredentialAPI, error)
```

Comment:
can we call this controllerCredentialAPIFunc
(to be consistent with the majority of the codebase, plus this PR has newCredentialAPIFunc for example)